scipy/sparse/linalg/eigen/_svds.py

```diff
@@ -216,18 +301,46 @@ def matmat_XH_X(x):
         if k == 1 and v0 is not None:
             X = np.reshape(v0, (-1, 1))
         else:
-            X = np.random.RandomState(52).randn(min(A.shape), k)
+            if rs_was_None:
+                X = np.random.RandomState(52).randn(min(A.shape), k)
```

Comment:
Can we use our `check_random_state` here?

Reply:
That's up in the previous function, `_iv` (input validation).

I don't understand how this fixed seed thing happened, but the purpose of this branching is to maintain the existing default behavior.

Reply:
ok, and it's not even 42 😅 Jokes aside can we put some deprecation warnings here?

Reply:
Definitely occurred to me, but I didn't want to do it piecemeal. There are a ton of things about `svds` that should be changed (e.g. positional arguments that do nothing for methods except `arpack`). I was thinking about creating a whole `scipy.sparse.svd` and deprecating `svds`, really.

Reply:
+1 for that and the name is also better IMO. To put on the imaginary list for 2.0 😅